### DISCUSSION TITLE
Mongodb Support in winter cms

### DISCUSSION THREAD

**Discussion by waleedbajwa1980** [wintercms/winter#1062] [General]
Hello, i have winter cms installed and built a website, now i want to connect it with mongo database. i have downloaded and installed its community version. i have installed mongodb support https://github.com/mongodb/laravel-mongodb/tree/3.9?tab=readme-ov-file#configuration
i have performed database migrations and seeds all done. but now hoe to use mongodb's Eloquent model , how to start in winter cms, kindly guide me.

**Comment by LukeTowers** [comment]
From Discord:

@bennothommo said:
> That is going to be extremely difficult to use with Winter. We already extend Eloquent for our own base model to add our required core functionality, which this plugin does as well. I mean, it's not impossible to create a plugin to implement this library, but you'd have to have some serious knowledge of Winter and Laravel's internals.
> I would strongly recommend sticking to Winter's supported database servers.

@waleedbajwa1980 said:
>  I have a scenario in which i have to receive about 900 data parameters on per minute basis, for that scenario i wanted to use mongdb with winter cms, otherwise i am pretty good with mysql. Can you suggest i should still use mysql or postgre or other supported dbs?

@bennothommo said:
> All the supported DBs can handle that amount of fields, it's just a bit of setting up first. You could also use the JSON type in MySQL to store entire objects if you want.
> Do you need to directly interact with all 900 fields?

@waleedbajwa1980 said: 
> Infact we have to collect three weather (temperature,humidity, air pressure) and six air pollutant params (SO2,NOX,CO,O3,PM2.5 and PM10) total 9*100 AQMS (air quality monitoring stations)=900 from 100 AQMS on per minue basis and has to maintain data for about 3 years. ALso there will be graphical dashboards based on the collected data params, complex statistical analysis etc mainly in the graphical form, graphs, charts, hardware calibration events etc Moreover custom dashboards and comparison charts with different time intervals from minute by minute upto year (1 minute, 15 minutes, hourly, daily, weekly, monthly , quartely and yearly) will also be part of the system.
> So do you consider storing data in mysql 's json datatype can be enough rather using mongodb for such large amount of data performing searching , comparing ranges of data etc!

@bennothommo said:
> I can't say whether MongoDB or MySQL's JSON field will be better for your use case - I don't know the extent of your data or what you intend to do with it. I'm saying that it's possible that the JSON field type in MySQL might be able to be used as a data storage for the entire data, but I would imagine anything you wish to filter, collate or calculate probably would be best as their own columns in MySQL. And as for MongoDB, it might be better for your use case, but Winter does not support MongoDB at this stage.

@josephcrowell said:
> I have a complex database tracking geo coordinates for around 250 delivery drivers each night. 10000-30000 records per driver each night. Data is kept for 6 months and Winter/MySql handles it fine. You just have to customize your settings to make MySql properly utilize the available resources from your hardware and make proper use of indexing
